src/main/java/io/cryostat/agent/remote/InvokeContext.java

```diff
@@ -0,0 +1,118 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.cryostat.agent.remote;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.lang.management.ManagementFactory;
+import java.util.Objects;
+
+import javax.inject.Inject;
+import javax.management.MBeanServer;
+import javax.management.ObjectName;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.sun.net.httpserver.HttpExchange;
+import org.apache.http.HttpStatus;
+import org.eclipse.microprofile.config.Config;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class InvokeContext extends MutatingRemoteContext {
+
+    private final Logger log = LoggerFactory.getLogger(getClass());
+    private final ObjectMapper mapper;
+
+    @Inject
+    InvokeContext(ObjectMapper mapper, Config config) {
+        super(config);
+        this.mapper = mapper;
+    }
+
+    @Override
+    public String path() {
+        return "/mbean-invoke/";
+    }
+
+    @Override
+    public void handle(HttpExchange exchange) throws IOException {
+        try {
+            String mtd = exchange.getRequestMethod();
+            switch (mtd) {
+                case "POST":
+                    try (InputStream body = exchange.getRequestBody()) {
+                        MBeanInvocationRequest req =
+                                mapper.readValue(body, MBeanInvocationRequest.class);
+                        if (!req.isValid()) {
+                            exchange.sendResponseHeaders(
+                                    HttpStatus.SC_BAD_REQUEST, BODY_LENGTH_NONE);
+                        }
+                        MBeanServer server = ManagementFactory.getPlatformMBeanServer();
+                        Object response =
+                                server.invoke(
+                                        ObjectName.getInstance(req.beanName),
+                                        req.operation,
+                                        req.parameters,
+                                        req.signature);
+                        if (Objects.nonNull(response)) {
+                            exchange.sendResponseHeaders(HttpStatus.SC_OK, BODY_LENGTH_UNKNOWN);
+                            try (OutputStream responseStream = exchange.getResponseBody()) {
+                                mapper.writeValue(responseStream, response);
+                            }
+                        } else {
+                            exchange.sendResponseHeaders(HttpStatus.SC_CREATED, BODY_LENGTH_NONE);
```

Comment:
```suggestion
                            exchange.sendResponseHeaders(HttpStatus.SC_ACCEPTED, BODY_LENGTH_NONE);
```

Thinking about it a bit more, I think 202 might be a better code for this condition than 201. Either way, it's a success code that is distinguishable from 200, so I don't think it's critically important. We might even want to return 201/202/204 depending on the exact operation being invoked later on, for example.